Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -363,19 +721,164 @@ void SkylanderPortalWindow::UpdateEdits()
+  {
+    if (character.game == IOS::HLE::USB::GIANTS)
+      pass = true;
+  }
+  if (m_game_filters[IOS::HLE::USB::SWAP_FORCE]->isChecked())
+  {
+    if (character.game == IOS::HLE::USB::SWAP_FORCE)
+      pass = true;
+  }
+  if (m_game_filters[IOS::HLE::USB::TRAP_TEAM]->isChecked())
+  {
+    if (character.game == IOS::HLE::USB::TRAP_TEAM)
+      pass = true;
+  }
+  if (m_game_filters[IOS::HLE::USB::SUPERCHARGERS]->isChecked())
+  {
+    if (character.game == IOS::HLE::USB::SUPERCHARGERS)
+      pass = true;
+  }
+  if (!pass)
+    return false;
+
+  // Check against search bar filter
+  if (!name.contains(m_sky_search->text(), Qt::CaseInsensitive))
+    return false;
+
+  // Check against active element filter
+  switch (GetElementRadio())
+  {
+  case 1:
+    if (character.element != IOS::HLE::USB::MAGIC)
+      return false;
+    break;
+  case 2:
+    if (character.element != IOS::HLE::USB::WATER)
+      return false;
+    break;
+  case 3:
+    if (character.element != IOS::HLE::USB::TECH)
+      return false;
+    break;
+  case 4:
+    if (character.element != IOS::HLE::USB::FIRE)
+      return false;
+    break;
+  case 5:
+    if (character.element != IOS::HLE::USB::EARTH)
+      return false;
+    break;
+  case 6:
+    if (character.element != IOS::HLE::USB::LIFE)
+      return false;
+    break;
+  case 7:
+    if (character.element != IOS::HLE::USB::AIR)
+      return false;
+    break;
+  case 8:
+    if (character.element != IOS::HLE::USB::UNDEAD)
+      return false;
+    break;
+  case 9:
+    if (character.element != IOS::HLE::USB::E_OTHER)
+      return false;
+    break;
+  }
+
+  return true;
 }
 
-void SkylanderPortalWindow::closeEvent(QCloseEvent* event)
+QString SkylanderPortalWindow::GetFilePath(u16 id, u16 var)
 {
-  hide();
+  QDir collection = QDir(m_collection_path);
+  auto& system = Core::System::GetInstance();
+  QString file_path;
+  for (auto file : collection.entryInfoList(QStringList(tr("*.sky"))))
+  {
+    File::IOFile sky_file(file.filePath().toStdString(), "r+b");
+    if (!sky_file)
+    {
+      continue;
+    }
+    std::array<u8, 0x40 * 0x10> file_data;
+    if (!sky_file.ReadBytes(file_data.data(), file_data.size()))
+    {
+      continue;
+    }
+    auto ids = system.GetSkylanderPortal().CalculateIDs(file_data);
+    if (ids.first == id && ids.second == var)
+    {
+      return file.filePath();
+    }
+  }
+  return file_path;
+}
+
+u8 SkylanderPortalWindow::GetCurrentSlot()
+{
+  for (auto radio : m_slot_radios)
+  {
+    if (radio->isChecked())
+    {
+      return radio->property("id").toInt();
+    }
+  }
+  return 0;
+}
+
+int SkylanderPortalWindow::GetElementRadio()
+{
+  for (auto radio : m_element_filter)
+  {
+    if (radio->isChecked())
+    {
+      return radio->property("id").toInt();
+    }
+  }
+  return -1;
+}
+
+QBrush SkylanderPortalWindow::GetBaseColor(std::pair<const u16, const u16> ids)
+{
+  switch (IOS::HLE::USB::list_skylanders.at(ids).game)
```

Comment:
This will crash the emulator if an invalid ID is given, which I think is possible by manually editing a skylanders file. Please change this to something like:

```
auto it = IOS::HLE::USB::list_skylanders.find(ids);
if (it == IOS::HLE::USB::list_skylanders.end()) {
  return QBrush(QColor(255, 255, 255, 255));
}

switch (it->second.game)
```